docs/modules/ROOT/pages/api/governance.adoc

```diff
@@ -667,3 +671,186 @@ Delegates votes from the sender to `delegatee`.
+* xref:#VotesComponent-delegates[`++delegates(self, account)++`]
+* xref:#VotesComponent-delegate[`++delegate(self, delegatee)++`]
+* xref:#VotesComponent-delegate_by_sig[`++delegate_by_sig(self, delegator, delegatee, nonce, expiry, signature)++`]
+--
+
+[.contract-index]
+.Internal implementations
+--
+.InternalImpl
+* xref:#VotesComponent-_delegate[`++_delegate(self, account, delegatee)++`]
+* xref:#VotesComponent-move_delegate_votes[`++move_delegate_votes(self, from, to, amount)++`]
+* xref:#VotesComponent-transfer_voting_units[`++transfer_voting_units(self, from, to, amount)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#VotesComponent-DelegateChanged[`++DelegateChanged(delegator, from_delegate, to_delegate)++`]
+* xref:#VotesComponent-DelegateVotesChanged[`++DelegateVotesChanged(delegate, previous_votes, new_votes)++`]
+--
+
+[#VotesComponent-Functions]
+==== Functions
+
+[.contract-item]
+[[VotesComponent-get_votes]]
+==== `[.contract-item-name]#++get_votes++#++(self: @ComponentState<TContractState>, account: ContractAddress) → u256++` [.item-kind]#external#
+
+Returns the current amount of votes that `account` has.
+
+[.contract-item]
+[[VotesComponent-get_past_votes]]
+==== `[.contract-item-name]#++get_past_votes++#++(self: @ComponentState<TContractState>, account: ContractAddress, timepoint: u64) → u256++` [.item-kind]#external#
+
+Returns the amount of votes that `account` had at a specific moment in the past.
+
+Requirements:
+
+- `timepoint` must be in the past.
+
+[.contract-item]
+[[VotesComponent-get_past_total_supply]]
+==== `[.contract-item-name]#++get_past_total_supply++#++(self: @ComponentState<TContractState>, timepoint: u64) → u256++` [.item-kind]#external#
+
+Returns the total supply of votes available at a specific moment in the past.
+
+NOTE: This value is the sum of all available votes, which is not necessarily the sum of all delegated votes.
+Votes that have not been delegated are still part of total supply, even though they would not participate in a
+vote.
+
+Requirements:
+
+- `timepoint` must be in the past.
+
+[.contract-item]
+[[VotesComponent-delegates]]
+==== `[.contract-item-name]#++delegates++#++(self: @ComponentState<TContractState>, account: ContractAddress) → ContractAddress++` [.item-kind]#external#
+
+Returns the delegate that `account` has chosen.
+
+[.contract-item]
+[[VotesComponent-delegate]]
+==== `[.contract-item-name]#++delegate++#++(ref self: ComponentState<TContractState>, delegatee: ContractAddress)++` [.item-kind]#external#
+
+Delegates votes from the sender to `delegatee`.
+
+Emits a {DelegateChanged} event.
+
+May emit one or two {DelegateVotesChanged} events.
+
+[.contract-item]
+[[VotesComponent-delegate_by_sig]]
+==== `[.contract-item-name]#++delegate_by_sig++#++(ref self: ComponentState<TContractState>, delegator: ContractAddress, delegatee: ContractAddress, nonce: felt252, expiry: u64, signature: Array<felt252>)++` [.item-kind]#external#
```

Comment:
+1 to using Span instead.

Reply:
When looking into this I realized we would need to change the type that `ISRC6::is_valid_signature` accepts(or convert from span to array - which doesn't make a lot of sense), and since it is part of the [SNIP-6](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-6.md) standard does not seem like the best idea either.
Thoughts on this? Should we leave it with Array?

Reply:
Similar to what is implemented in SRC9, we should accept Span and just call `.into()` for is_valid_signature, and this should be less expensive than receiving `Array<felt252>` instead, so we don't have a reason to compromise more interfaces when we should be prioritizing `Span<felt252>` over `Array<felt252>` in external function inputs (because of the deserialization required for dispatchers).